lib/matplotlib/axes/_axes.py

```diff
@@ -3205,7 +3207,14 @@ def get_next_color():
         slices = []
         autotexts = []
 
-        for frac, label, expl in zip(x, labels, explode):
+        if isinstance(wedgeprops, collections.abc.Sequence):
+            if len(wedgeprops) != len(x):
+                raise ValueError(f'wedgeprops length={len(wedgeprops)}'
+                                 f' and input length={len(x)} are not equal')
+        else:
+            wedgeprops = itertools.repeat(wedgeprops)
+
```

Comment:
Just a small side note here: for some other cases (like colorbars with hatches, which is what I primarily thought of), the corresponding properties will cycle rather than error if they are not the same. Not sure what is preferred though.